sdk/eventhub/azure-eventhubs/azure/eventhub/eventprocessor/event_processor.py

```diff
@@ -0,0 +1,171 @@
+            call user defined PartitionProcessor.process_events()
+        """
+        log.info("EventProcessor %r is being started", self._instance_id)
+        client = self._eventhub_client
+        partition_ids = await client.get_partition_ids()
+        self.partition_ids = partition_ids
+        claimed_list = await self._claim_partitions()
+        await self._start_claimed_partitions(claimed_list)
+
+    async def stop(self):
+        """Stop all the partition consumer
+
+        It sends out a cancellation token to stop all partitions' EventHubConsumer will stop receiving events.
+
+        """
+        self._cancellation_token.cancel()
+        # It's not agreed whether a partition manager has method close().
+        log.info("EventProcessor %r cancellation token has been sent", self._instance_id)
+
+    async def _claim_partitions(self):
+        partitions_ownership = await self._partition_manager.list_ownership(self._eventhub_name, self._consumer_group_name)
+        partitions_ownership_dict = dict()
+        for ownership in partitions_ownership:
+            partitions_ownership_dict[ownership["partition_id"]] = ownership
+
+        to_claim_list = []
+        for pid in self.partition_ids:
+            p_ownership = partitions_ownership_dict.get(pid)
+            if p_ownership:
+                to_claim_list.append(p_ownership)
+            else:
+                new_ownership = dict()
+                new_ownership["eventhub_name"] = self._eventhub_name
+                new_ownership["consumer_group_name"] = self._consumer_group_name
+                new_ownership["instance_id"] = self._instance_id
+                new_ownership["partition_id"] = pid
+                new_ownership["owner_level"] = 1  # will increment in preview 3
+                to_claim_list.append(new_ownership)
+        claimed_list = await self._partition_manager.claim_ownership(to_claim_list)
+        return claimed_list
+
+    async def _start_claimed_partitions(self, claimed_partitions):
+        consumers = []
+        for partition in claimed_partitions:
+            partition_id = partition["partition_id"]
+            offset = partition.get("offset")
+            offset = offset or self._initial_event_position
+            consumer = self._eventhub_client.create_consumer(self._consumer_group_name, partition_id,
+                                                            EventPosition(str(offset)))
+            consumers.append(consumer)
+
+            partition_processor = self._partition_processor_callable(
+                eventhub_name=self._eventhub_name,
+                consumer_group_name=self._consumer_group_name,
+                partition_id=partition_id,
+                checkpoint_manager=CheckpointManager(partition_id, self._eventhub_name, self._consumer_group_name,
+                                                     self._instance_id, self._partition_manager)
+            )
+            loop = asyncio.get_running_loop()
```

Comment:
`asyncio.get_running_loop()` was introduced in 3.7.

Reply:
Changed to use a utility function
```python
def get_running_loop():
    try:
        return asyncio.get_running_loop()
    except AttributeError:  # 3.5 / 3.6
        loop = asyncio._get_running_loop()  # pylint: disable=protected-access
        if loop is None:
            raise RuntimeError('No running event loop')
        return loop
```